docs/v1/P4Runtime-Spec.mdk

```diff
@@ -1644,15 +1644,26 @@ observed_value = server.read(p4_entity)
 assert(intended_value == observed_value)
 ~ End Pseudo
 
-To ensure read-write symmetry, the rest of this document tries to offer canonical
-representations for various data types, but this principle should be thought of
-where it falls short. Ensuring this will allow client software to recover
-programmatically from failures that can affect the switch stack software,
-communication channel, or the client replicas. If `Read` RPC returns a
+To ensure read-write symmetry, the rest of this document tries to offer
+canonical representations for various data types, but this principle should be
+thought of where it falls short. Ensuring this will allow client software to
+recover programmatically from failures that can affect the switch stack
+software, communication channel, or the client replicas. If `Read` RPC returns a
 semantically-same but syntactically-different response then the client would
 have to canonicalize the read values to check its internal state, which only
 pushes the protocol's complexities to the client implementations.
 
+In order to avoid placing too much burden on the P4Runtime server
+implementation, we do not in general mandate that the order of values in a
+Protobuf repeated field be preserved. For example, the server is not required to
```

Comment:
Do we want to qualify this with something like "in a repeated field that is anywhere inside of an `Entity` message"?  I believe this comment doesn't really apply to repeated fields used outside of `Entity` messages.

Reply:
Could you clarify what you mean by that? What kind of repeated field outside of `Entity` do you have in mind (that can be both used in a WriteRequest and a ReadResponse)?

Reply:
Maybe it is already qualified somewhere near here, but there are several repeated fields defined in p4info.proto and p4runtime.proto that can never appear inside of an `Entity` message, and thus also cannot appear inside of a WriteRequest or ReadResponse message, and this text isn't saying anything about whether order is significant or insignificant for those, is it?

Reply:
Given that this section as a whole only talks about Write & Read, and therefore entities, I'm not sure where the confusion is coming from, but I don't mind adding that clarification. Are you referring to the updates repeated field used to send batches for example? Because this field never appears in a ReadResponse, and there is no real symmetry to talk about.

Reply:
Re-reading the beginning of the section I am commenting on here, I think it is fine to commit this as is now, without further clarification.  It does seem to properly qualify which repeated fields it could possibly be talking about to write and read requests.  Thanks.